src/proxy/http/remap/PluginDso.cc

```diff
@@ -282,15 +282,15 @@ PluginDso::clean(std::string &error)
 void
 PluginDso::acquire()
 {
-  ++this->_instanceCount;
-  PluginDbg(_dbg_ctl(), "plugin DSO acquire (ref-count:%d, dso-addr:%p)", this->_instanceCount.load(), this);
+  this->refcount_inc();
+  PluginDbg(_dbg_ctl(), "plugin DSO acquire (ref-count:%d, dso-addr:%p)", this->refcount(), this);
```

Comment:
Instead of reverting the use of _intanceCount (although not deleting _instanceCount),would it make more sense to delete the RefCountObjInHeap base class?
```
$ git diff
diff --git a/include/proxy/http/remap/PluginDso.h b/include/proxy/http/remap/PluginDso.h
index e7c84e97c..2fefd208a 100644
--- a/include/proxy/http/remap/PluginDso.h
+++ b/include/proxy/http/remap/PluginDso.h
@@ -50,7 +50,7 @@ namespace fs = swoc::file;
 
 #include "proxy/Plugin.h"
 
-class PluginThreadContext : public RefCountObjInHeap
+class PluginThreadContext
 {
 public:
   virtual void                       acquire() = 0;
$ 
$ cmake --build build
[0/2] Re-checking globbed directories...
ninja: no work to do.
$ 
```
RefCountObjInHeap is meant as a base clases for classes whose instance lifetimes will be controlled by instances of Ptr (defined in  include/tscore/Ptr.h).

Reply:
I reverted back to what it was before as with the new code it was failing to properly count  DSO's and remap [plugins](https://docs.trafficserver.apache.org/en/latest/developer-guide/design-documents/reloading-plugins.en.html#plugindso). (two different counters)

```
-class PluginThreadContext : public RefCountObjInHeap
```

we will not be able to call:
```
refcount_inc()
refcount_dec()
refcount()
```
as used in the code.

we can replace this by another `std::atomic` but I didn't want to make more changes.

Reply:
Then why did the code compile when I removed RefCountInHeap as a base class above?

Reply:
did it?
```
$ cmake --build build
[0/2] Re-checking globbed directories...
ninja: no work to do. <<<<<<<<<<
```

I've got:
```
/home/xxx/code/git/trafficserver/src/proxy/http/remap/PluginDso.cc: In member function ‘virtual void PluginDso::acquire()’:
/home/xxx/code/git/trafficserver/src/proxy/http/remap/PluginDso.cc:285:9: error: ‘class PluginDso’ has no member named ‘refcount_inc’
  285 |   this->refcount_inc();
      |         ^~~~~~~~~~~~
In file included from /home/xxx/code/git/trafficserver/include/tscore/Diags.h:36,
...
/home/xxx/code/git/trafficserver/src/proxy/http/remap/PluginDso.cc:286:81: error: ‘class PluginDso’ has no member named ‘refcount’
  286 |   PluginDbg(_dbg_ctl(), "plugin DSO acquire (ref-count:%d, dso-addr:%p)", this->refcount(), this);
```

Reply:
Because your PR changned `++this->_instanceCount'`to `this->refcount_int()` .

Reply:
:rofl: 
well ...  yes.